oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderTests.java

```diff
@@ -524,6 +573,30 @@ public void decodeWhenCacheIsConfiguredAndValueLoaderErrorsThenThrowsJwtExceptio
 				.hasMessageContaining("An error occurred while attempting to decode the Jwt");
 	}
 
+	// gh-8730
+	@Test
+	public void withJwkSetUriWhenUsingCustomTypeHeaderThenRefuseOmittedType() throws Exception {
```

Comment:
I had to "swap" the logic in the test withJwkSetUriWhenUsingCustomTypeHeaderThenRefuseOmittedType for the following reason: `NimbusJwtDecoderTests.JWK_SET` does not match `TestKeys.DEFAULT_PRIVATE_KEY`, which makes it impossible to construct a signed JWT with custom header typ. If you have a better idea I am happy to improve/align this test with the other two.

Reply:
I think this is fine. Note that the major goal of a test like this is just to make sure that your customization is used. So Mockito would be fine in this case, too; just call `verify` at the end of the test.

Reply:
I opted against Mockito since it doesn't guarantee the `jwtProcessor` passed to the callback is actually the one used to verify (aka passed to `NimbusJwtDecorder`). Code like `jwtProcessorCustomizer.accept(new DefaultJWTProcessor())` would verify just fine with Mockito.